packages/netlify-cms-backend-azure/src/API.ts

```diff
@@ -0,0 +1,1007 @@
+      },
+    ];
+
+    await this.requestJSON({
+      method: 'POST',
+      url: `${this.endpointUrl}/refs`,
+      body: JSON.stringify(deleteBranchPayload),
+    });
+  }
+
+  uploadAndCommit(
+    items: AzureCommitItem[],
+    comment = 'Creating new files',
+    branch: string = this.branch,
+    newBranch = false,
+  ) {
+    return this.getRef(branch).then(async (ref: AzureRef) => {
+      if (ref == null) {
+        ref = await this.getRef(this.branch);
+      }
+
+      if (newBranch) {
+        ref = new AzureRef(this.branchToRef(branch), ref.objectId);
+      }
+
+      const commit = new AzureCommit(comment);
+
+      items.forEach((i: AzureCommitItem) => {
+        switch (i.action as AzureCommitChangeType) {
+          case AzureCommitChangeType.ADD:
+            commit.changes.addBase64(i.path, i.base64Content);
+            break;
+          case AzureCommitChangeType.EDIT:
+            commit.changes.editBase64(i.path, i.base64Content);
+            break;
+          case AzureCommitChangeType.DELETE:
+            commit.changes.delete(i.path);
+            break;
+          case AzureCommitChangeType.RENAME:
+            commit.changes.rename(i.path, i.path);
+            break;
+        }
+      });
+
+      // Only bother with a request if we're going to make changes.
+      if (commit.changes.length > 0) {
+        const push = new AzurePush(ref);
+        push.commits.push(commit);
+
+        return this.requestJSON({
+          url: `${this.endpointUrl}/pushes`,
+          method: 'POST',
+          headers: { 'Content-Type': 'application/json; charset=utf-8' },
+          body: JSON.stringify(push),
+        });
+      }
+    });
+  }
+
+  async readUnpublishedBranchFile(contentKey: string) {
+    const { branch, collection, slug, path, status, mediaFiles } = await this.retrieveMetadata(
+      contentKey,
+    );
+    const [fileData, isModification] = await Promise.all([
+      this.readFile(path, null, { branch }) as Promise<string>,
+      this.isFileExists(path, this.branch),
+    ]);
+
+    return {
+      slug,
+      metaData: { branch, collection, objects: { entry: { path, mediaFiles } }, status },
+      fileData,
+      isModification,
+    };
+  }
+
+  async retrieveUnpublishedEntryData(contentKey: string) {
+    const { collection, slug } = parseContentKey(contentKey);
+    const branch = this.branchFromContentKey(contentKey);
+    const pullRequest = await this.getBranchPullRequest(branch);
+    const diffs = await this.getDifferences(pullRequest.sourceRefName);
+    const diffsWithIds = await Promise.all(
+      diffs.map(async d => {
+        const path = d.item.path;
+        const newFile = d.changeType === 'add';
+        const id = d.item.objectId;
+        return { id, path, newFile };
+      }),
+    );
+    const label = pullRequest.labels.find(l => isCMSLabel(l.name, this.cmsLabelPrefix));
+    const labelName = label && label.name ? label.name : DEFAULT_NETLIFY_CMS_LABEL_PREFIX;
```

Comment:
We should only be reading pull requests with a CMS label attached to them.
Not sure when `label` can be undefined, and in any case we shouldn't default to `DEFAULT_NETLIFY_CMS_LABEL_PREFIX`